src/envoy/http/authn/authn_utils.cc

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "authn_utils.h"
+#include "common/common/base64.h"
+#include "common/json/json_loader.h"
+
+namespace Envoy {
+namespace Http {
+namespace Istio {
+namespace AuthN {
+
+bool AuthnUtils::GetJWTPayloadFromHeaders(
+    const HeaderMap& headers, const LowerCaseString& jwt_payload_key,
+    ::google::protobuf::Map< ::std::string, ::std::string>* claims) {
+  const HeaderEntry* entry = headers.get(jwt_payload_key);
+  if (!entry) {
+    ENVOY_LOG(error, "{}: no JwtPayloadKey entry {} in the header",
+              __FUNCTION__, jwt_payload_key.get());
+    return false;
+  }
+  std::string value(entry->value().c_str(), entry->value().size());
+  std::string payload_str = Base64::decode(value);
```

Comment:
you need to use Base64URLDecode()  which is different with Base64::decode()

Reply:
Done.